python/GafferSceneTest/RenderControllerTest.py

```diff
@@ -527,5 +527,217 @@ def testNullObjects( self ) :
 		controller.setMinimumExpansionDepth( 10 )
 		controller.update()
 
+	def testBlur( self ) :
+
+		sphere = GafferScene.Sphere()
+		sphere["type"].setValue( sphere.Type.Primitive )
+		sphereFilter = GafferScene.PathFilter()
+		sphereFilter["paths"].setValue( IECore.StringVectorData( [ "/sphere" ] ) )
+		sphereAttributes = GafferScene.StandardAttributes()
+		sphereAttributes["in"].setInput( sphere["out"] )
+		sphereAttributes["filter"].setInput( sphereFilter["out"] )
+
+		group = GafferScene.Group()
+		group["in"][0].setInput( sphereAttributes["out"] )
+		groupFilter = GafferScene.PathFilter()
+		groupFilter["paths"].setValue( IECore.StringVectorData( [ "/group" ] ) )
+		groupAttributes = GafferScene.StandardAttributes()
+		groupAttributes["in"].setInput( group["out"] )
+		groupAttributes["filter"].setInput( groupFilter["out"] )
+
+		options = GafferScene.StandardOptions()
+		options["in"].setInput( groupAttributes["out"] )
+
+		# Animated source for testing
+		frame = GafferTest.FrameNode()
+
+		# Source that isn't animated, but has an animated hash
+		dummyFrame = Gaffer.Node()
+		dummyFrame["output"] = Gaffer.FloatPlug()
+		dummyFrame["expression"] = Gaffer.Expression()
+		dummyFrame["expression"].setExpression( 'parent["output"] = context.getFrame() * 0 + 3' )
+
+		renderer = GafferScene.Private.IECoreScenePreview.CapturingRenderer()
+		controller = GafferScene.RenderController( options["out"], Gaffer.Context(), renderer )
+		controller.setMinimumExpansionDepth( 2 )
+		controller.update()
+
+		def assertMotionSamples( sampleTimes, deform ) :
+
+			capturedSphere = renderer.capturedObject( "/group/sphere" )
+			self.assertIsNotNone( capturedSphere )
+
+			if deform:
+				samples = [ i.radius() for i in capturedSphere.capturedSamples() ]
+				times = capturedSphere.capturedSampleTimes()
+				#for i, r in enumerate( sampleTimes ) :
+					#self.assertEqual( samples[i].radius(), r )
+			else:
+				samples = [ i.translation().x for i in capturedSphere.capturedTransforms() ]
+				times = capturedSphere.capturedTransformTimes()
+				#for i, r in enumerate( sampleTimes ) :
+					#self.assertEqual( samples[i].radius(), r )
+
+			self.assertEqual( len( samples ), len( sampleTimes ) )
+			for (i,j) in zip( samples, sampleTimes ):
+				self.assertAlmostEqual( i, j, places = 6 )
```

Comment:
Not particularly surprising to me - I'm using some samples that are 0.2 apart, which isn't exactly representable.  Splitting the range [0.6, 1.4] into quarters I would definitely expect to introduce some sort of floating point error, and we're probably not being particularly careful about it.  Why wouldn't you expect things to be off by an epsilon at least?

Reply:
I was thinking it wouldn't matter what the error was, because the values we're sampling are driven exactly by `context.getFrame()`, so should have the exact same value as was used for the shutter open/close. But you're right, the expected values are coming from Python, where they're computed differently _and_ represented as doubles.